src/purchases.ts

```diff
@@ -666,6 +675,25 @@ export default class Purchases {
     RNPurchases.syncObserverModeAmazonPurchase(productID, receiptID, amazonUserID, isoCurrencyCode, price);
   }
 
+  /**
+   * Use this method only if you already have your own IAP implementation using StoreKit 2 and want to use
+   * RevenueCat's backend. If you are using StoreKit 1 for your implementation, you do not need this method.
+   *
+   * You only need to use this method with *new* purchases. Subscription updates are observed automatically.
+   *
+   * Important: This should only be used if you have enabled observer mode during SDK configuration.
+   *
+   * @warning You need to finish the transaction yourself after calling this method.
+   *
+   * @param {string} productID Product ID that was just purchased
+   * @returns {Promise<PurchasesStoreTransaction>} If there was a transacton found and handled for the provided product ID.
+   */
+  public static async handleObserverModeTransaction(productID: string): Promise<PurchasesStoreTransaction> {
+    await Purchases.throwIfAndroidPlatform();
+    await Purchases.throwIfNotConfigured();
+    return RNPurchases.handleObserverModeTransactionForProductID(productID);
```

Comment:
Hmm something I'm noticing is that this may return a null value in iOS <15... We either need to make the return type nullable or return an error (I don't think we can return a transaction).

I think returning an error might make more sense in this case?

Reply:
Switched to returning an error